students/yushus/session02/series.py

```diff
@@ -0,0 +1,67 @@
+
+def fibonacci(n):
+    """ compute the nth Fibonacci number """
+
+    if n < 0:
+        raise "Input number has to be non-negative!!!"
+
+    if n <= 1:
+        return n
+    else:
+        return (fibonacci(n-1) + fibonacci(n-2))
+
+
+
+def lucas(n):
+    """ compute the nth Lucas number """
+
+    if n < 0:
+        raise "Input number has to be non-negative!!!"
+
+    if n == 0:
+        return 2
+    elif n == 1:
+        return 1
+    else:
+        return (lucas(n-1) + lucas(n-2))
+
+def sum_series(n, n0=0, n1=1):
+    """
+    compute the nth value of a summation series.
+
+    :param n0=0: value of zeroth element in the series
+    :param n1=1: value of first element in the series
+
+    if n0 == 0 and n1 == 1, the result is the Fibbonacci series
+
+    if n0 == 2 and n1 == 1, the result is the Lucas series
+    """
+    if n0 == 0 and n1 == 1:
+        return fibonacci(n)
+    elif n0 == 2 and n1 == 1:
+        return lucas(n)
+    else:
+        raise "Not the right format!!!"
```

Comment:
This is not correct. The `sum_series()` function should be applicable for any `n0` and `n1` values, not just for Fibonacci or Lucas series. Basically your `sum_series()` function is expected to call itself recursively so that it'll calculate any sum series, including Fibonacci and Lucas series. The key point is that your `sum_series()` should return `n0` if `n==0`, or return `n1` if `n==1`, or return the corresponding recursive result. Please fix.

Reply:
Thank you so much Hosung. I see my mistake now :-). Changed.